mike_skalandunas/app/js/tapirs/controllers/tapirs_controller.js

```diff
@@ -0,0 +1,60 @@
+'use strict';
+
+module.exports = function(app) {
+  app.controller('TapirsController', ['$scope', '$http', function($scope, $http) {
+    $scope.tapirs = [];
+    var defaults = {flavor: 'Malayan', diet: 'veggies'};
+    $scope.newTapir = Object.create(defaults);
+
+    $scope.getAll = function() {
+      $http.get('/api/tapirs')
+        .then(function(res) {
+          $scope.tapirs = res.data;
+        }, function(err) {
+          console.log(err.data);
+        });
+    }
+
+    $scope.create = function(tapir) {
+      $http.post('/api/tapirs', tapir)
+        .then(function(res) {
+          $scope.tapirs.push(res.data);
+          $scope.newTapir = Object.create(defaults);
+        }, function(err) {
+          console.log(err.data)
+        });
+    };
+
+    $scope.update = function(tapir) {
+      tapir.editing = false;
+      $http.put('/api/tapirs/' + tapir._id, tapir)
+        .then(function(res) {
+          console.log('The tapir\'s name has been changed.');
+        }, function(err) {
+          $scope.errors.push('could not get tapir: ' + tapir.name + '.');
+        });
+    };
+
+    $scope.remove = function(tapir) {
+      $scope.tapirs.splice($scope.tapirs.indexOf(tapir), 1);
+      $http.delete('/api/tapirs/' + tapir._id)
+        .then(function(res) {
+          console.log('Tapir\'s shipped off to Portland.');
+        }, function(err) {
+          console.log(err.data);
+          $scope.errors.push('Shipment of ' + tapir.name + ' cancelled.');
+          $scope.getAll();
+        });
+    };
+
+    $scope.reset = function(tapir) {
+      $scope.master = {};
+
+      $scope.reset = function() {
+        angular.copy($scope.master, $scope.user);
+      };
+
+      $scope.reset();
```

Comment:
This doesn't look like it works right. I would assume, you would need to set $scope.master equal to a copy of the $scope.tapirs after the api call. Then a reset is just setting the edited array equal to the master.